sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -524,22 +524,41 @@ class Dataset[T] private[sql](
    */
   @Experimental
   @InterfaceStability.Evolving
-  def checkpoint(): Dataset[T] = checkpoint(eager = true)
+  def checkpoint(eager: Boolean = true): Dataset[T] = _checkpoint(eager = eager)
```

Comment:
I think we should keep this signature  - there's already a ` def checkpoint(eager: Boolean = true)`

Reply:
I always try to avoid duplication of code, and with docs this takes ~10 lines for nothing - I believe a function with a default parameter is as readable as the function without the parameter. But please let me know if it goes against the code style. Thanks

Reply:
No, it's going to break Java API compatibility. Default value does not work with Java.